### DISCUSSION TITLE
ERROR: Duplicate entry for key 'CLIENT_ATTRIBUTES.PRIMARY'

### DISCUSSION THREAD

**Discussion by aholguin508** [keycloak/keycloak#15489] [Archive]
Keycloak is not getting successful upgraded because it is showing an error that says ERROR: Duplicate entry 'b0d06c23-d9e0-4db0-a136-072ac415ae92-post.logout.redirect.uris' for key 'CLIENT_ATTRIBUTES.PRIMARY'..  The upgrade is from keycloak 15 to 19.. Any idea on this ?

**Comment by lexcao** [comment]
Could you please try to query this to see the duplicated data?
```
SELECT * FROM CLIENT_ATTRIBUTES  WHERE 
client_id = 'b0d06c23-d9e0-4db0-a136-072ac415ae92'
AND name = 'post.logout.redirect.uris';
```

Have you manually added this attribute `post.logout.redirect.uris`?
If not, remove one of them and restart to try to upgrade again.

**Reply by aholguin508** [reply]
I was expecting that if an info is already in DB, it skips because it already exists.. I deleted it and rerun the upgrade and it works. Thank you!!
